Slim/Player/Protocols/HTTP.pm

```diff
@@ -52,6 +52,60 @@ sub new {
 	return $self;
 }
 
+sub request {
+	my $self = shift;
+	my $args  = shift;
+	my $song = $args->{'song'};
+	my $track = $song->track;
+
+	# no other guidance, define AudioBlock to make sure that audio_offset is skipped in requestString
+	if (!defined $track->initial_block_type || $song->stripHeader) {
+		$song->initialAudioBlock('');
+		return $self->SUPER::request($args);
+	}
+
+	# obtain initial audio block if missing and adjust seekdata then
+	if (!defined $song->initialAudioBlock && Slim::Formats->loadTagFormatForType($track->original_content_type)) {
+		my $formatClass = Slim::Formats->classForFormat($track->original_content_type);
+		my $seekdata = $song->seekdata || {};
+		
+		if ($formatClass->can('findFrameBoundaries')) {
+			my $offset = $formatClass->findFrameBoundaries($track->initial_block_fh, $seekdata->{sourceStreamOffset} || 0, $seekdata->{timeOffset} || 0);
+			$seekdata->{sourceStreamOffset} = $offset if $offset;
+		}
+		
+		if ($formatClass->can('getInitialAudioBlock')) {
+			my $initialBlock = \$formatClass->getInitialAudioBlock($track->initial_block_fh, $track, $seekdata->{timeOffset} || 0);
+			$song->initialAudioBlock($$initialBlock);
+		}	
```

Comment:
Is there too much referencing/de-referencing going on? `getInitialAudioBlock()` would return a scalar. No need to de-reference it?

Reply:
I was not sure about that. I added an interim variable for readability, but I'm still a bit confused for when Perl actually copies string (bytes). I was under the impression that $initialBlock = $format... would create a copy of the block

Reply:
For example,
```
my $a = "somestring";
my $b = $a;
$a = substr($a, 0, 2);
print("$b\n$a\n");
```
Here I have "somestring" and "so" for for me Perl makes $b a copy of $a. This is different for hash references

Reply:
Scalars usually are treated as copies, where references obviously are not. 

But in the above code you seem to create a reference to a non-referenced response (coming from `getInitialAudioBlock()`), just to de-reference it one line further down again. I believe in this case it's neither required nor useful.

Reply:
Do you know how I could check if, when I do
```my $initialBlock = \$formatClass->getInitialAudioBlock($track->initial_block_fh, $track, $seekdata->{timeOffset} || 0);
$song->initialAudioBlock($$initialBlock);
```
The scalar initialBlock does not receive a copy of the getInitialAudioBlock result?

Reply:
That's correct, in this case I'd expect `$initialBlock` to be a copy indeed. But TBH I don't know whether your construct would change this. Wouldn't that just give you a reference to the returned value, which you then de-reference, creating that other copy you wanted to prevent?

Reply:
I've moved to a direct call, without using an interim reference. It still readable ...